mlir/lib/Dialect/Rock/Transforms/AlignTiling.cpp

```diff
@@ -258,64 +226,24 @@ static Value reconfigureLAGeneric(PatternRewriter &b,
   return laOut;
 }
 
-static LogicalResult findGlobalStore(linalg::GenericOp laGeneric,
-                                     Value &inputLeadingToGlobalStore,
-                                     GlobalStoreOp &gemmStoreOp) {
-  for (auto pair :
-       llvm::zip(laGeneric.inputs(), laGeneric.getIndexingMapsArray())) {
-    AffineMap inpIdxMap = std::get<1>(pair);
-    Value input = std::get<0>(pair);
-    GlobalStoreOp maybeStore = traceToGlobalStore(input);
-    if (maybeStore) {
-      if (gemmStoreOp) {
-        LLVM_DEBUG(llvm::dbgs()
-                   << "Multiple generic inputs come from writeback\n");
-        return failure();
-      }
-
-      auto laGenericOut = laGeneric.getOutputOperand(0);
-      auto laGenericOutIdxMap = laGeneric.getTiedIndexingMap(laGenericOut);
-      auto invertOutIdxMap = inversePermutation(laGenericOutIdxMap);
-      auto outToInMap = inpIdxMap.compose(invertOutIdxMap);
-      SmallVector<unsigned> permutedDims;
-      // This is not to allow broadcasting but due to canonical linalg
-      // form if the unit dims can carry affine expr to be zero in the
-      // translation, hence there is a following check.
-      if (!outToInMap.isMinorIdentityWithBroadcasting(&permutedDims)) {
-        LLVM_DEBUG(llvm::dbgs() << outToInMap << "\n");
-        LLVM_DEBUG(
-            llvm::dbgs()
-            << "The store is not even a minor identity with broadcasting.\n");
-        return failure();
-      }
-      auto inpShape = input.getType().cast<ShapedType>().getShape();
-      LLVM_DEBUG(llvm::dbgs() << "outToInMap = " << outToInMap << "\n");
-      LLVM_DEBUG(llvm::dbgs() << "inp shape = "
-                              << input.getType().cast<ShapedType>() << "\n");
-      LLVM_DEBUG(llvm::dbgs() << "permutedDims = ");
-      LLVM_DEBUG(llvm::interleaveComma(permutedDims, llvm::dbgs()));
-      LLVM_DEBUG(llvm::dbgs() << "\n");
-
-      for (auto bDim : permutedDims) {
-        if (inpShape[bDim] != 1) {
-          LLVM_DEBUG(llvm::dbgs()
-                     << "The store input cannot be a real broacast.\n");
-          return failure();
-        };
+static Value findThreadwiseWrite(linalg::GenericOp laGeneric,
```

Comment:
nit : I'd like this to return LogicalResult rather than a null in the case of failure...

Reply:
You'd be looking for `FailureOr<Value>` but also, null as the failure value for Value/Attribute/... is reasonably accepted

Reply:
I'd prefer FailureOr<Value> over null -- It seems easy to miss to check the null, IMHO where as in missing failure check gives out a compiler warning.

Anyway its nit picking (and subjective)...

Reply:
For this case, it is not a failure to not find the ThreadwiseWriteAll, and I prefer to return a null object for simple conditional test and usage.